### ISSUE TITLE
It seems WorkbenchFX is not SpringBoot compatible!

### ISSUE THREAD

**Issue by linspinho** [dlsc-software-consulting-gmbh/WorkbenchFX#604]
When I "insert" WorkbenchFX in my Project it does work!
Is there a way to get it to work?

**Comment by dlemmermann** [comment]
This is the worst ticket I have seen in a long time. This is like calling an IT support hotline and saying "my computer doesn't work, what's wrong?". The ticket is unspecific and contains no error messages.

**Comment by linspinho** [comment]
You're right! My bad! Maybe my mistake was to put this problem in ISSUES, I should have put this in DISCUSSION. 
Anyway, I have been working on for several days, I have done so many tries that I don't have a code to show, but basically in my Main class I starts springboot and initWorkbench() that initiates customModule(){} class that initiates the Node by Controller, no matter if it is FXML or JavaFX code UI it does not start a Spring service @autowired annotated in that Controller. just wants to say that sample code I use works before deploying WorkbenchFX.

**Comment by mapleafly** [comment]
use [javafxweaver](https://github.com/rgielen/javafx-weaver)
note: springboot base 2.6.x

**Comment by balazsvero** [comment]
If your controller is not a spring bean, don't expect it to process spring annotations. Actually, there are a few nice examples of springboot and javafx integrations and I'm here only because I'm working on a prototype application for future projects using spring boot (3.*) + javafx as well, and found WorkbenchFX as a nice UI for it. Far from being ready to even be published, but it works fine. Start your spring app in javafx application's init method: 
` @Override
    public void init() {
        SpringApplication springApplication = new SpringApplication(SpringApp.class);
        springApplication.setBannerMode(Banner.Mode.OFF);
        springApplication.setLogStartupInfo(false);
        this.context = springApplication.run();
    }`

Then if you need a spring bean to be the controller of a javafx component, get its reference via spring's DI. E.g. set controllerFactory on the FXMLLoader and set the controller class in the FXML. 
`    private FXMLLoader getFXMLLoader() {
        FXMLLoader fxmlLoader = new FXMLLoader();
        fxmlLoader.setControllerFactory(clz -> context.getBean(clz));
        return fxmlLoader;
    }`

If a bean is a javafx component, that is used more than once in your application, don't forget to make it a PROTOTYPE bean, so each time it is injected, a new instance will be created.
You can make all your WorkbenchModules spring beans and then you can use any spring feature you want.
